streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -0,0 +1,363 @@
+        String topicCTestMessage = "topic-C test";
+        String topicYTestMessage = "topic-Y test";
+        String topicZTestMessage = "topic-Z test";
+
+
+        final Serde<String> stringSerde = Serdes.String();
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        KStream<String, String> pattern1Stream = builder.stream(Pattern.compile("topic-\\d"));
+        KStream<String, String> pattern2Stream = builder.stream(Pattern.compile("topic-[A-D]"));
+        KStream<String, String> namedTopicsStream = builder.stream(TOPIC_Y, TOPIC_Z);
+
+        pattern1Stream.to(stringSerde, stringSerde, DEFAULT_OUTPUT_TOPIC);
+        pattern2Stream.to(stringSerde, stringSerde, DEFAULT_OUTPUT_TOPIC);
+        namedTopicsStream.to(stringSerde, stringSerde, DEFAULT_OUTPUT_TOPIC);
+
+        KafkaStreams streams = new KafkaStreams(builder, streamsConfiguration);
+        streams.start();
+
+        Properties producerConfig = getProducerConfig();
+
+        IntegrationTestUtils.produceValuesSynchronously(TOPIC_1, Arrays.asList(topic1TestMessage), producerConfig);
+        IntegrationTestUtils.produceValuesSynchronously(TOPIC_2, Arrays.asList(topic2TestMessage), producerConfig);
+        IntegrationTestUtils.produceValuesSynchronously(TOPIC_A, Arrays.asList(topicATestMessage), producerConfig);
+        IntegrationTestUtils.produceValuesSynchronously(TOPIC_C, Arrays.asList(topicCTestMessage), producerConfig);
+        IntegrationTestUtils.produceValuesSynchronously(TOPIC_Y, Arrays.asList(topicYTestMessage), producerConfig);
+        IntegrationTestUtils.produceValuesSynchronously(TOPIC_Z, Arrays.asList(topicZTestMessage), producerConfig);
+
+        Properties consumerConfig = getConsumerConfig();
+
+        List<String> expectedReceivedValues = Arrays.asList(topicATestMessage, topic1TestMessage, topic2TestMessage, topicCTestMessage, topicYTestMessage, topicZTestMessage);
+        List<KeyValue<String, String>> receivedKeyValues = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerConfig, DEFAULT_OUTPUT_TOPIC, 6);
+        List<String> actualValues = new ArrayList<>(6);
+
+        for (KeyValue<String, String> receivedKeyValue : receivedKeyValues) {
+            actualValues.add(receivedKeyValue.value);
+        }
+
+        streams.close();
+        Collections.sort(actualValues);
+        Collections.sort(expectedReceivedValues);
+        assertThat(actualValues, equalTo(expectedReceivedValues));
+    }
+
+    //TODO should be updated to expected = TopologyBuilderException after KAFKA-3708
+    @Test(expected = AssertionError.class)
+    public void testNoMessagesSentExceptionFromOverlappingPatterns() throws Exception {
+
+        String fooMessage = "fooMessage";
+        String fMessage = "fMessage";
+
+
+        final Serde<String> stringSerde = Serdes.String();
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        /*
+          overlapping patterns here, no messages should be sent as TopologyBuilderException
```

Comment:
Nit: we can just use `//` for in-function comments, without capitalizing the first letter, and start with a space after `//`.